types/webpack-env/index.d.ts

```diff
@@ -103,21 +104,21 @@ declare namespace __WebpackModuleApi {
          * The data will be available at module.hot.data on the new module.
          * @param callback
          */
-        dispose<T>(callback: (data: T) => void): void;
+        dispose<T = any>(callback: (data: T) => void): void;
```

Comment:
I mean, neither should `require` either; if you're making `require`'s type argument optional, you might as well just make it non-generic in the first place.